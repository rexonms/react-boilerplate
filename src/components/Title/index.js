import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';

const Title = props => (
  <div>
    <h1 className={`title ${props.type}`}>{props.label}</h1>
  </div>
);

Title.defaultProps = {
  type: 'default',
};

Title.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.string,
};

export default Title;
