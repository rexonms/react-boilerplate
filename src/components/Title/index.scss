.title {
  color: #ccc;
}

.small {
  font-size: 18px;
}

.default {
  font-size: 24px;
}

.large {
  font-size: 32px;
}

