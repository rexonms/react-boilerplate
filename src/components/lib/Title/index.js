import React from 'react';
import PropTypes from 'prop-types';

import Styles from './Title.scss';

const propTypes = {
  /** Text for title */
  title: PropTypes.string.isRequired,
};

const defaultProps = {};

const Title = props => (
  <div className={`${Styles.container}`}>
    <h2>{props.title}</h2>
  </div>
);

Title.propTypes = propTypes;
Title.defaultProps = defaultProps;

export default Title;
