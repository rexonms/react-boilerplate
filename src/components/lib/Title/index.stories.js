import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select, number } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import Title from './index';

const stories = storiesOf('components/lib/Title', module);

stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('Title', withInfo(`
  Component
  ~~~js
    <Title name='Title' />
  ~~~
`)(() => {
  const vehicles = ['Model S', 'Model 3', 'Model X', 'Roadster 2'];
  const vehicleOptions = select('vehicle', vehicles, vehicles[0]);
  const defaultValue = 5;
  const options = {
    range: true,
    min: 1,
    max: 5,
    step: 1,
  };
  const groupId = 'GROUP-ID1';
  const vehicleCounts = number('count', defaultValue, options, groupId);
  return (
    <Title
      vehicle={vehicleOptions}
      count={vehicleCounts}
    />
  );
}));