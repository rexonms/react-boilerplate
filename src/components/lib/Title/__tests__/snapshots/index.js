import React from 'react';
import { shallow } from 'enzyme';

import Title from '../../index';

it('Renders component correctly', () => {
  const props = {
    title: 'Momo Eating Contest',
  };
  const component = shallow(<Title {...props} />);
  expect(component).toMatchSnapshot();
});
