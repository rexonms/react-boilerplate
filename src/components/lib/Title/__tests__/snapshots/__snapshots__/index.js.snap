// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title
    title="Momo Eating Contest"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        Momo Eating Contest
</h2>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Momo Eating Contest",
      },
      "ref": null,
      "rendered": "Momo Eating Contest",
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Momo Eating Contest
</h2>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Momo Eating Contest",
        },
        "ref": null,
        "rendered": "Momo Eating Contest",
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
